src/utils/pickRevisionItem.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode } from "@azure/arm-appcontainers";
+import { AzureWizardPromptStep, ContextValueQuickPickStep, IActionContext, QuickPickWizardContext, ResourceGroupsItem, runQuickPickWizard } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { ContainerAppItem } from "../tree/ContainerAppItem";
+import { RevisionItem } from "../tree/revisionManagement/RevisionItem";
+import { RevisionsItem } from "../tree/revisionManagement/RevisionsItem";
+import { localize } from "./localize";
+import { PickItemOptions, pickContainerApp } from "./pickContainerApp";
+
+interface RevisionPickItemOptions extends PickItemOptions {
+    // Automatically select a RevisionItem without re-prompting the user
+    selectByRevisionName?: string;
+}
+
+function getPickRevisionStep(tdp: vscode.TreeDataProvider<unknown>): AzureWizardPromptStep<QuickPickWizardContext> {
+    return new ContextValueQuickPickStep(tdp, {
+        contextValueFilter: { include: RevisionItem.contextValueRegExp },
+        skipIfOne: true,
+    });
+}
+
+export function getPickRevisionsStep(tdp: vscode.TreeDataProvider<unknown>): AzureWizardPromptStep<QuickPickWizardContext> {
+    return new ContextValueQuickPickStep(tdp,
+        {
+            contextValueFilter: { include: RevisionsItem.contextValueRegExp },
+            skipIfOne: true,
+        },
+        {
+            placeHolder: localize('selectRevisionItem', 'Select a revision item')
+        });
+}
+
+/**
+ * Without re-prompting, automatically selects a RevisionItem matching the known 'revisionName'
+ */
+async function selectWithoutPrompt(startingNode: ContainerAppItem | RevisionsItem, revisionName: string): Promise<RevisionItem> {
```

Comment:
It would be better to use something like a ContextValueQuickPickStep or create a subclass of GenericQuickPickStep that matches this and does `skipIfOne=true`.